src/api/jest/index.ts

```diff
@@ -34,6 +34,25 @@ type DefaultOptions =
  */
 export const mergePreset = <
   AdditionalOptions extends keyof Config.InitialOptions,
->(
-  options: Pick<Config.InitialOptions, AdditionalOptions | DefaultOptions>,
-): Config.InitialOptions => mergeRaw(jestPreset, options);
+>({
+  projects,
+  ...options
+}: Pick<
+  Config.InitialOptions,
+  AdditionalOptions | DefaultOptions
+>): Config.InitialOptions => ({
+  ...mergeRaw(jestPreset, options),
+  projects: projects?.map((project) => {
+    if (typeof project === 'string') {
+      return project;
+    }
+
+    return {
+      transform: jestPreset.transform,
+      moduleNameMapper: jestPreset.moduleNameMapper,
+      ...project,
+    };
+  }),
+});
```

Comment:
Thoughts on the following options:

```suggestion
>): Config.InitialOptions => {
  const root = mergeRaw(jestPreset, options);

  return {
    ...root,

    projects: projects?.map((project) => {
      if (typeof project === 'string') {
        return project;
      }

      return {
        ...root,
        ...project,
      };
    }),
  };
};
```

```suggestion
>): Config.InitialOptions => {
  const root = mergeRaw(jestPreset, options);

  return {
    ...root,

    projects: projects?.map((project) => {
      if (typeof project === 'string') {
        return project;
      }

      return {
        moduleNameMapper: root.moduleNameMapper,
        transform: root.transform,
        ...project,
      };
    }),
  };
};
```

Though curiously the [documentation](https://jestjs.io/docs/configuration#projects-arraystring--projectconfig) states the following; is this is meant to work OOTB, or potentially only for the `string` option?

> With the `projects` option enabled, Jest will copy the root-level configuration options to each individual child configuration during the test run, resolving its values in the child's context. This means that string tokens like `<rootDir>` will point to the _child's root directory_ even if they are defined in the root-level configuration.

Reply:
Going with the second. With the first you get alot of warnings complaining about unknown config so it appears only a subset of settings actually work for the `projects` key

Reply:
![image](https://github.com/seek-oss/skuba/assets/18017094/87c570cd-38da-45cc-9875-9f996fee369d)

Just screenshot for my memory if we ever wonder why again